src/coreclr/jit/helperexpansion.cpp

```diff
@@ -1171,3 +1178,331 @@ bool Compiler::fgExpandStaticInitForCall(BasicBlock** pBlock, Statement* stmt, G
     call->gtInitClsHnd = NO_CLASS_HANDLE;
     return true;
 }
+
+//------------------------------------------------------------------------------
+// fgVNBasedIntrinsicExpansion: Expand specific calls marked as intrinsics using VN.
+//
+// Returns:
+//    PhaseStatus indicating what, if anything, was changed.
+//
+PhaseStatus Compiler::fgVNBasedIntrinsicExpansion()
+{
+    PhaseStatus result = PhaseStatus::MODIFIED_NOTHING;
+
+    if (!doesMethodHaveSpecialIntrinsics() || opts.OptimizationDisabled())
+    {
+        return result;
+    }
+
+    // TODO: Replace with opts.compCodeOpt once it's fixed
+    const bool preferSize = opts.jitFlags->IsSet(JitFlags::JIT_FLAG_SIZE_OPT);
+    if (preferSize)
+    {
+        // The optimization comes with a codegen size increase
+        JITDUMP("Optimized for size - bail out.\n")
+        return result;
+    }
+    return fgExpandHelper<&Compiler::fgVNBasedIntrinsicExpansionForCall>(true, true);
+}
+
+//------------------------------------------------------------------------------
+// fgVNBasedIntrinsicExpansionForCall : Expand specific calls marked as intrinsics using VN.
+//
+// Arguments:
+//    block - Block containing the intrinsic call to expand
+//    stmt  - Statement containing the call
+//    call  - The intrinsic call
+//
+// Returns:
+//    True if expanded, false otherwise.
+//
+bool Compiler::fgVNBasedIntrinsicExpansionForCall(BasicBlock** pBlock, Statement* stmt, GenTreeCall* call)
+{
+    assert(call->gtCallMoreFlags & GTF_CALL_M_SPECIAL_INTRINSIC);
```

Comment:
```suggestion
    assert((call->gtCallMoreFlags & GTF_CALL_M_SPECIAL_INTRINSIC) != 0);
```

according to coding conventions